torch/csrc/autograd/python_function.cpp

```diff
@@ -340,8 +340,61 @@ static void _wrap_outputs(const std::shared_ptr<PyNode>& cdata, THPFunction *sel
     }
   }
 
+  _jvp_fn_t jvp_user_function = [self](variable_list inputs, variable_list grad_inputs) {
+    pybind11::gil_scoped_acquire gil;
```

Comment:
We actually release the GIL when executing expensive c++ code.
In this case, I think we do keep it during the post-processing.
But this function is sent to some pure c++ code. And since that code is not aware of the GIL, I prefer to have it again here.